src/lib/Iterator.class.ts

```diff
@@ -27,7 +27,7 @@ class Iterator extends EventEmitter {
   private readonly query: SelectQuery;
   public readonly endpoint: Endpoint;
   private readonly engine: QueryEngine;
-  private readonly delay: number | undefined;
+  private readonly delay: number = 0;
```

Comment:
Keep types simple if you can: no union type needed here (`| undefined`) so remove it.